hazelcast/src/test/java/com/hazelcast/map/EntryProcessorTest.java

```diff
@@ -844,6 +844,23 @@ public void testMapEntryProcessorPartitionAware() {
         assertEquals(1, instance2.getMap(mapName2).get(1));
     }
 
+    @Test
+    public void testMapEntryProcessorPartitionAwareSameMap() {
```

Comment:
We need to delete this kind of tests which are doing remote call inside EP. Especially the ones doing back calls. Because it is not a supported feature, it causes dead-locks. Maybe we can invent a check to prevent such usages. Only doing map.get like calls inside an EP can be ok to use for now. But having such usages are generally bad practice and can make system stuck.

Reply:
so in entry processor for given IMap you cannot update this map but can read from it? If so, I can change the test to perform get in entry processor, it also should trigger the issue in TS

Reply:
Only gets on local partition are ok, so you need to be sure that the key you pass map.get is in the current partition that your EP is running, otherwise deadlock is also valid for get. 
I think the best way is not to allow any such calls from EP.

Reply:
there are some checks regarding nested operations in `MapOperation.isMapStoreOffloadEnabled()`, invoked by `EntryProcessor.innerBeforeRun()`